homeworks/yuriy.chaban_YuriyChaban/homework_2/hw2.js

```diff
@@ -0,0 +1,65 @@
+const readline = require('readline');
+
+const rl = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout,
+});
+
+const rangesArray = [];
+let numberItem;
+let reversedNumber;
+let rightDigit;
+let minRange;
+let maxRange;
+let isMinIsNumber;
+let isMaxIsNumber;
+let isMaxAndMinNumber;
```

Comment:
The way how you implement this is very similar to using global variables, and it tends to complex errors. On another hand, you could use a more functional approach, when functions are pure - get value, calculate and return the result.

Reply:
```
let r;

// function with side effect
function sum(a, b) {
     r = a + b;
}

// pure function
function sum(a, b) {
     return a + b;
}
```

pure functions much easier to test, and they more predictable.